src/cdn_definitions/data.yaml

```diff
@@ -251,3 +251,61 @@ repo_overrides:
       if_created_after: "2023-06-01T14:00:00Z"
       key: example
       value: true
+
+# directory_layouts is used to define DirectoryLayout objects and consist of a
+# pattern, to match against content URL, and attributes for the resulting
+# object, often present as regex groups in the pattern.
+#
+# Allowed value types; string, bool, null
+#
+# With the exception of "rhelalt", which is expected to be stored as bool,
+# a "true" value is used to signal that regex matching should be applied for the
+# corresponding group in the pattern.
+#
+# E.g.;
+#   `type: true`    ==  `type = re.match(...).group("type")`
+#   `type: false`   ==  `type = False`
+#   `type: "dist"`  ==  `type = "dist"`
+directory_layouts:
+  - name: DirectoryLayoutC
+    # "/content/<type>/<platform>/<rhui>/<variant>/<major>.<minor>/<arch>/<rest>"
+    pattern: "/(?:content|shadow)/(?P<type>aus|dist|beta|(?:retired-)?els|eus|htb|rhb|tus|e4s)/(?P<platform>rhes|rhs|cf-me)/(?P<rhui>rhui/)?(?P<variant>[^/]+)/(?P<major_version>[0-9])\\.(?P<minor_version>[0-9]+)/(?P<arch>[^/]+)/(?P<rest>.+)$"
+    type: true
+    platform: true
+    rhui: true
+    variant: true
+    major_version: true
+    minor_version: true
+    arch: true
+    rest: true
+  - name: DirectoryLayoutD
+    # "/content/<type>/<platform>/<rhui>/<variant>/<major>/<minor>/<arch>/<rest>"
+    pattern: "(?P<up_to_minor>/(?:content|shadow)/(?P<type>aus|dist|beta|(?:retired-)?els|eus|fastrack|htb|rhb|tus|e4s)/(?P<platform>cf-me|rhel|rhevh|rhes|rhs)/(?P<rhui>rhui/)?(?P<variant>[^/]+)/(?P<major_version>[0-7][^/]*))/(?P<minor_version>(?:\\$releasever|\\$[0-9]+|[34][AEW]S|[34]Desktop))/(?P<arch>[^/]+)/(?P<rest>.+)$"
```

Comment:
Personally I think it might be a good idea to establish the convention of starting the patterns with `^`.

Currently it seems ambiguous whether the patterns are meant to be used with `re.search` (find anywhere in a string) or `re.match` (find at beginning of string).  Based on context I believe only re.match makes sense, but a casual reader might not realize that.  It seems to invite bugs where someone will take these patterns and feed them into re.search.

If the patterns start with `^` then they should behave the same whether used with `re.search` or `re.match`, protecting against that kind of bug.

Also, revisiting the topic of breaking the patterns over multiple lines, this is an example where one could make use of indenting to improve readability.  There is a rather large capture group here covering multiple segments:

```yaml
pattern: |-
  (?x)^
  (?P<up_to_minor>
      /
      (?:content|shadow)/
      (?P<type>aus|dist|beta|(?:retired-)?els|eus|fastrack|htb|rhb|tus|e4s)/
      (?P<platform>cf-me|rhel|rhevh|rhes|rhs)/
      (?P<rhui>rhui/)?
      (?P<variant>[^/]+)/
      (?P<major_version>[0-7][^/]*)
  )
  /
  (?P<minor_version>(?:\$releasever|\$[0-9]+|[34][AEW]S|[34]Desktop))/
  (?P<arch>[^/]+)/
  (?P<rest>.+)$
```

Reply:
Yeah, agreed, this is good. Thanks!